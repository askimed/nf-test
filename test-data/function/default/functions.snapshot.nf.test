nextflow_function {

  name "Test Function Say Hello"

  script "test-data/function/default/functions.nf"
  function "say_hello"

  test("Passing case") {

    when {
      function {
        """
        input[0] = "aaron"
        """
      }
    }

    then {
      snapshot(function).match("forer lukas")
      //assert function.success
      //assert function.result == "Hello aaron"
      //assert function.stdout.contains("Hello aaron") 
      //assert function.stderr.isEmpty()
    }

  }

}