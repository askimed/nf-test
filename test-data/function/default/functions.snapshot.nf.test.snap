{
    "forer lukas": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": null,
                "exitStatus": 0,
                "failed": false,
                "result": "Hello aaron",
                "stdout": [
                    "Hello aaron"
                ],
                "errorMessage": null,
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 0,
                    "tasksSucceeded": 0
                },
                "name": "say_hello",
                "success": true
            }
        ],
        "timestamp": "2023-08-26T15:20:15.069134"
    }
}